### PR TITLE
[4.0] Give com_modules a service interface so that com_config can integrate

### PR DESCRIPTION
**Pull request by wilsonge**
Fixes issue in com_config saving modules 

```
Too few arguments ModulesController.php on line 106 and exactly 3 expected
```

See https://github.com/joomla/joomla-cms/issues/20354 for full testing instructions

### CODE REVIEW

File: administrator/components/com_modules/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use Joomla\CMS\Extension\ComponentInterface;
+use Joomla\CMS\Extension\Service\Provider\DispatcherFactory;
+use Joomla\CMS\Extension\Service\Provider\MVCFactoryFactory;
+use Joomla\CMS\MVC\Factory\MVCFactoryFactoryInterface;
+use Joomla\Component\Modules\Administrator\Extension\ModulesComponent;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+
+/**
+ * The content service provider.
```
Comment (laoneo) [inline administrator/components/com_modules/services/provider.php]: Its the module provider

**Review by laoneo — CHANGES_REQUESTED** [review]

File: administrator/components/com_modules/services/provider.php (hunk 1 above)
Comment (wilsonge) [reply administrator/components/com_modules/services/provider.php]: Oops! Caught on my copy pasting :)

**Comment by infograf768** [comment]
I have not tested this item.<br /><br />This solves https://github.com/joomla/joomla-cms/issues/20354<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20382">issues.joomla.org/tracker/joomla-cms/20382</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 422aa35f06ce14984b0daf7080d9d7cfb57de36d<br /><br />This solves https://github.com/joomla/joomla-cms/issues/20354<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20382">issues.joomla.org/tracker/joomla-cms/20382</a>.</sub>

**Review by laoneo — APPROVED** [review]

**Review by laoneo — APPROVED** [review]

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 422aa35f06ce14984b0daf7080d9d7cfb57de36d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20382">issues.joomla.org/tracker/joomla-cms/20382</a>.</sub>

**Pull request merged**